### DISCUSSION TITLE
Callabcks or other notification upon workflow completion

### DISCUSSION THREAD

**Discussion by atararaksin** [Netflix/conductor#3636] [General]
Hi team,
In my system, Conductor workflows will be triggered by an external system. To avoid having to poll for workflow statuses, I'd like to have Conductor either send a callback to the triggering system, or to publish a message to a queue like Rabbit. Is this possible and is there any recommendation for this?

Looking at the [docs](https://conductor.netflix.com/documentation/configuration/eventhandlers.html#event-task) it seems like using Event Task that would either perform a callback or publish a message is one way to implement it. My concerns here are:
- If the workflow fails during one of the subsequent steps, the Event task will not be triggered and there will be no notification about workflow failure. Any way to work around?
- Ideally I'd want the notification to contain the payload with the final state of workflow, including outputs, status (completed/failed), etc. If the notification is sent during the last step of workflow, technically the workflow will not yet be completed at that moment.

**Comment by charybr** [comment]
Hi,

I have created a PR for Callback. Basically this is a Http callback (webhook).
https://github.com/Netflix/conductor-community/pull/228 - Task Status notification
https://github.com/Netflix/conductor-community/pull/222 - Workflow status notification

Yet to get approval and merge it.

**Reply by v1r3n** [reply]
Thanks @charybr taking a look.
